lib/cartopy/io/img_tiles.py

```diff
@@ -378,6 +391,60 @@ def find_images(self, target_domain, target_z, start_tile=None):
                 yield self.tms_to_quadkey(tile, google=True)
 
 
+class TileCache(object):
+    """Simple caching API"""
+
+    def put(self, url, img):
+        """
+        :param url: key used in cache
+        :param img: value used in cache
+        """
+        pass
+
+    def get(self, url):
+        """
+        :param url: key to lookup in cache
+        """
+        pass
+
+    def close(self):
+        pass
+
+
+class ShelveTileCache(TileCache):
+    def __init__(self, cache_name=None):
+        """
+        Cache implementation using shelve.
+
+        If appdirs is available, the proper user_cache_dir for the system will
+        be used, otherwise the current working directory will be used.
```

Comment:
I'm not sure that's the best place; there's a `config` setting that might be a better default.